src/yaml.rs

```diff
@@ -55,6 +66,42 @@ pub enum Yaml {
 pub type Array = Vec<Yaml>;
 pub type Hash = LinkedHashMap<Yaml, Yaml>;
 
+#[derive(Clone, PartialEq, PartialOrd, Debug, Eq, Ord, Hash)]
+pub struct CommentedYaml(pub Box<Yaml>, pub Comments);
+
+#[derive(Clone, PartialEq, PartialOrd, Debug, Eq, Ord, Hash)]
```

Comment:
```suggest
#[derive(Clone, PartialEq, PartialOrd, Debug, Eq, Ord, Hash, Default)]
```
It would be nice to have a default implementation given there are so many fields.